File: Introduccion/ejercicio 5/Empleado.java
package introduccion;

public class Empleado {
    private String nombre;
    private String horario;
    
    public Empleado(String nombre, String horario){
        this.nombre = nombre;
        this.horario = horario;
        System.out.println("Empleado creado: "+ nombre + ", Horario="+ horario);
    }
    public String getNombre(){
        return nombre;
    }
    public void setNombre(String nombre){
        this.nombre = nombre;
    }
    public String getHorario(String horario){
        return horario;
    }
    public void setHorario(String horario){
        this.horario = horario;
    }
    @Override
    public String toString(){
        return "Empleado [Nombre=" + nombre + ", Horario=" + horario + "]";
    } 
    @Override
    protected void finalize () throws Throwable{
        System.out.println("Empleado eliminado: " + this.nombre);
        super.finalize();
    }
}

File: Introduccion/ejercicio 5/Empleado.py
class Empleado:
    def __init__(self, nombre, horario):
        self.nombre = nombre
        self.horario = horario
        print(f"Empleado creado: {self.nombre}, Horario:{self.horario}")

    def get_nombre(self):
        return self.nombre
    def set_nombre(self):
        self.nombre = nombre
    def get_horario(self):
        return self.horario
    def set_horario(sefl):
        self.horario = horario
    def __str__(self):
        return f"Empleado [Nombre={self.nombre}, Horario={self.horario}]"
    def __del__(self):
        print(f"Empleado eliminado:{self.nombre}")

if __name__ == "__main__":
    empleado1 = Empleado ("Ana", "8:00 - 16:00")
    print(empleado1)

    del empleado1
